esp-wifi/src/wifi/mod.rs

```diff
@@ -1367,24 +1298,13 @@ pub(crate) mod embassy {
         where
             F: FnOnce(&mut [u8]) -> R,
         {
-            let res = critical_section::with(|cs| {
-                let mut queue = DATA_QUEUE_TX.borrow_ref_mut(cs);
-
-                let mut packet = unwrap!(
-                    DataFrame::new(),
-                    "unreachable: transmit()/receive() ensures there is a buffer free and space available"
-                );
+            WIFI_TX_INFLIGHT.fetch_add(1, Ordering::SeqCst);
```

Comment:
Would it be possible to deduplicate this method with the other impl TxToken? Then maybe it wouldn't be hard to turn the `tmp` array into a local static and avoid the big stack frame, depending on what the project prefers.

Reply:
I moved the impl into a function for tx  & rx, which are called in both the embassy and smoltcp consume impls